Indexing/SubstitutionTree_FastGen.cpp

```diff
@@ -19,20 +19,20 @@
 #include "Kernel/Matcher.hpp"
 #include "Kernel/SubstHelper.hpp"
 
-#include "SubstitutionTree.hpp"
-
 namespace Indexing
 {
 
 
 
-const unsigned SubstitutionTree::GenMatcher::BACKTRACK_SEPARATOR;
+template<class LeafData_>
+const unsigned SubstitutionTree<LeafData_>::GenMatcher::BACKTRACK_SEPARATOR;
```

Comment:
tbh i don't know what this does. I just added the template mechanically so it will compiled. My intuition is the following:
During generalization we keep a stack of var -> term assignments. We need to be able to backtrack these, a bunch at a time. The special constant `BACKTRACK_SEPARATOR` helps us to mark which bunches we want to backtrack at once. But I don't understand what's going on in `*_Fast{Inst,Gen}` in details so I'd rather not add a semi-correct comment ;) 

We need the templating here as `SubstitutionTree` is not a `class` but a `template<class> class`. Otherwise we cannot specify the path. And you're right that this is surely being inlined ways.

Reply:
Sounds sensible to me. Thanks!